presto-main/src/main/java/com/facebook/presto/operator/LongArrayBlockEncodingBuffers.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import io.airlift.slice.SliceOutput;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.Arrays;
+
+import static com.facebook.presto.operator.ByteArrayUtils.writeLengthPrefixedString;
+import static com.google.common.base.Verify.verify;
+import static io.airlift.slice.SizeOf.SIZE_OF_INT;
+import static io.airlift.slice.SizeOf.sizeOf;
+import static java.lang.Math.max;
+import static sun.misc.Unsafe.ARRAY_LONG_INDEX_SCALE;
+
+public class LongArrayBlockEncodingBuffers
+        extends BlockEncodingBuffers
+{
+    private static final String NAME = "LONG_ARRAY";
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(LongArrayBlockEncodingBuffers.class).instanceSize();
+
+    private byte[] valuesBuffer;
+    private int valuesBufferIndex;
+
+    public LongArrayBlockEncodingBuffers(int initialBufferPositions)
+    {
+        super(initialBufferPositions);
+        prepareBuffers();
+    }
+
+    @Override
+    protected void prepareBuffers()
+    {
+        valuesBuffer = new byte[initialBufferPositions * ARRAY_LONG_INDEX_SCALE];
+        valuesBufferIndex = 0;
+    }
+
+    @Override
+    public void resetBuffers()
+    {
+        bufferedPositionCount = 0;
+        valuesBufferIndex = 0;
+        resetNullsBuffer();
+    }
+
+    @Override
+    public void accumulateRowSizes(int[] rowSizes)
+    {
+        int averageElementSize = Long.BYTES + Byte.BYTES;
+
+        for (int i = 0; i < positionCount; i++) {
+            rowSizes[i] += averageElementSize;
+        }
+    }
+
+    @Override
+    protected void accumulateRowSizes(int[] positionOffsets, int positionCount, int[] rowSizes)
+    {
+        int averageElementSize = Long.BYTES + Byte.BYTES;
```

Comment:
- since this is an exact value and not an approximation, I'd remove `average` from the name and use a static constant
- elsewhere, `positionOffsets` is an array of positionCount + 1 elements where i-th element specifies the offset of the i-th entry in the array; if you follow this convention, then the logic can be simplified 

```
private static final int POSITION_SIZE = Long.BYTES + Byte.BYTES;

        for (int i = 0; i < positionCount; i++) {
            rowSizes[i] += (positionOffsets[i + 1] - positionOffsets[i]) * POSITION_SIZE;
        }
```